saml2/saml2-service-provider/src/opensaml4Main/java/org/springframework/security/saml2/provider/service/web/authentication/OpenSaml4AuthenticationRequestResolver.java

```diff
@@ -55,8 +55,10 @@ public OpenSaml4AuthenticationRequestResolver(RelyingPartyRegistrationResolver r
 	@Override
 	public <T extends AbstractSaml2AuthenticationRequest> T resolve(HttpServletRequest request) {
 		return this.authnRequestResolver.resolve(request, (registration, authnRequest) -> {
-			authnRequest.setIssueInstant(Instant.now(this.clock));
+			Instant issuingInstant = Instant.now(this.clock);
```

Comment:
I appreciate the extra effort to add a timing-based validation, though I'm concerned that this may be too opinionated as it will necessarily introduce timeout configurations. Would you mind if we stay focused on `InResponseTo` for now? You can open a separate ticket to discuss the timeout feature, if you like.

Reply:
Cool, the related problem to `InRespnseTo` is that if we assume that the existence of a saved request indicates that we should validate `InResponseTo` in the response, then a saved stale request efficiently and improperly blocks IdP initiated logins (e.g., we first try to log in using SP initiated login which is for some reason aborted with the saved request still in the backend associated with our cookie, 15 minutes later we go to the IdP and try to login via a link to our app, now the app will reject the login attempt because there is a stale stored request which indicated that we should validate `InResponseTo` in the login attempt which is missing since it is a IdP initiated login. An alternative would be to only validate `InResponseTo` if it is present in the response but that kind of circumvents one of the main ideas behind it, namely that we should know when to expect an `InResponseTo` and when not. You think we should just ignore this situation and remove the eviction?

Reply:
Good points. I think that if there is no `InResponseTo`, then validation should pass. You can see `OpenSamlLogoutResponseValidator#validateInResponseTo` for an example.

`InResponseTo` helps mitigate replay attacks, but this validation doesn't apply if the IdP is configured to do unsolicited login. So, if `InResponseTo` is missing, there's no problem unless it is evidence of misconfiguration. If it's misconfiguration, the IdP needs to be changed to not allow unsolicited responses and the SP needs to be customized to always reject missing `InResponseTo` responses.